src/dbnode/storage/index.go

```diff
@@ -2073,6 +2075,26 @@ func (i *nsIndex) unableToAllocBlockInvariantError(err error) error {
 	return ierr
 }
 
+func (i *nsIndex) SetExtendedRetentionPeriod(period time.Duration) {
```

Comment:
Instead of an explicit call to SetExtendedRetentionPeriod, would it not be better to just model this in the namespace proto perhaps? That way it can be instantiated on creation time on the reverse index for the namespace, and if none is explicitly set for the index (an independent retention period), then it can default to the retention of the namespace (as it currently does today).